lib/rucio/tests/test_api_external_representation.py

```diff
@@ -128,3 +161,257 @@ def test_api_account_limit(self):
+
+        # test scope_list
+        out = scope_list(self.scope_name, recursive=True, **self.vo)
+        for did in out:
+            assert_equal(did['scope'], self.scope_name)
+            if did['parent'] is not None:
+                assert_equal(did['parent']['scope'], self.scope_name)
+
+        # test get_did
+        add_did_to_followed(self.scope_name, 'ext_parent', self.account_name, **self.vo)
+        out = get_users_following_did('ext_parent', self.scope_name, **self.vo)
+        for user in out:
+            assert_equal(user['user'], self.account_name)
+
+    def test_api_exporter(self):
+        """ EXPORTER (API): Test external representation of exported data """
+
+        out = export_data('root', **self.new_vo)
+        rses = out['rses']
+        assert_in(self.rse3_name, rses)
+        assert_not_in(self.rse3_id, rses)
+
+        distances = out['distances']
+        assert_in(self.rse3_name, distances)
+        assert_not_in(self.rse3_id, distances)
+        assert_in(self.rse4_name, distances[self.rse3_name])
+        assert_not_in(self.rse4_id, distances[self.rse3_name])
+
+        # check for interference from other VOs
+        if self.multi_vo:
+            assert_not_in(self.rse_name, rses)
+            assert_not_in(self.rse_id, rses)
+            assert_not_in(self.rse2_name, rses)
+            assert_not_in(self.rse2_id, rses)
+
+            assert_not_in(self.rse_name, distances)
+            assert_not_in(self.rse_id, distances)
+            assert_not_in(self.rse2_name, distances)
+            assert_not_in(self.rse2_id, distances)
+
+    def test_api_identity(self):
+        """ IDENTITY (API): Test external representation of identity accounts """
+
+        id_key = ''.join(random.choice(string.ascii_lowercase) for x in range(10))
+
+        add_account_identity(id_key, 'userpass', self.account_name, 'rucio_test@test.com', 'root', default=True, password='ext_pass', **self.vo)
+
+        # TODO - fix passing default to add_account_identity
+        # out = get_default_account(id_key, 'userpass')
+        # assert_equal(out, self.account_name)
+
+        out = list_accounts_for_identity(id_key, 'userpass')
+        assert_in(self.account_name, out)
+        if self.multi_vo:
+            assert_not_in(self.account.internal, out)
+
+    def test_api_replica(self):
+        """ REPLICA (API): Test external representation of replicas """
+
+        did = 'ext_' + str(generate_uuid())
+        add_replicas('MOCK', [{'scope': self.scope_name, 'name': did, 'bytes': 100}], issuer='root', **self.vo)
+
+        add_did(self.scope_name, 'ext_parent_2', 'dataset', issuer='root', account=self.account_name, **self.vo)
+        attachment = {'scope': self.scope_name, 'name': 'ext_parent_2',
+                      'dids': [{'scope': self.scope_name, 'name': did}]}
+        attach_dids_to_dids([attachment], issuer='root', **self.vo)
+
+        # get_did_from_pfns - in progress but requires other fixes
```

Comment:
Likewise, does this need a separate issue.

Reply:
This is dependent on #33 since I had issues with internal/external scopes in the pfns. I'll add these comments to the code, I meant to do it before the initial commit but forgot.